### PR TITLE
add tutorial for building your own chatbot

### PR DESCRIPTION
**Pull request by ylwu-amzn**
### Description
Add tutorial for how to build your own chatbot in OpenSearch with Agent Framework.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+        ],
+        "model_id": "xkJLyI0Bn3OCesyvf94S",
+        "embedding_field": "stock_price_history_embedding",
+        "doc_size": 3
+      }
+    }
+  ]
+}
+```
+Test with
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:577]:
```suggestion
Test by running the agent:
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 2]
```diff
@@ -0,0 +1,836 @@
...
+      "url": "https://api.openai.com/v1/chat/completions",
+      "headers": {
+        "Authorization": "Bearer ${credential.openAI_key}"
+      },
+      "request_body": "{ \"model\": \"${parameters.model}\", \"messages\": [{\"role\":\"system\",\"content\":\"${parameters.system_instruction}\"},{\"role\":\"user\",\"content\":\"${parameters.prompt}\"}] }"
+    }
+  ]
+}
+```
+Create model with connector id in response:
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:621]:
```suggestion
Create a model using the connector ID from the response:
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 3]
```diff
@@ -0,0 +1,836 @@
...
+POST /_plugins/_ml/models/your_openai_model_id/_predict
+{
+  "parameters": {
+    "system_instruction": "You are an Assistant which can answer kinds of questions.",
+    "prompt": "hello"
+  }
+}
+```
+
+Create Agent with custom `system_instruction` and `prompt`.
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:642]:
```suggestion
Create an agent with custom `system_instruction` and `prompt`.
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 4]
```diff
@@ -0,0 +1,836 @@
...
+
+`prompt` includes these placeholders (read more details in step 4.1): `tool_descriptions`, `chat_history`, `format_instruction`, `question` and `scratchpad`
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:644]:
```suggestion
The `prompt` includes the following placeholders (for more information, see step 4.1): `tool_descriptions`, `chat_history`, `format_instruction`, `question`, and `scratchpad`.
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 5]
```diff
@@ -0,0 +1,836 @@
...
+        "model_id": "your_embedding_model_id_from_step1",
+        "embedding_field": "stock_price_history_embedding",
+        "doc_size": 3
+      }
+    }
+  ]
+}
+```
+
+Note agent id from response, execute agent to test:
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:701]:
```suggestion
Note the agent ID from the response and test the model by running the agent:
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 6]
```diff
@@ -0,0 +1,836 @@
...
+```
+POST _plugins/_ml/agents/your_agent_id/_execute
+{
+  "parameters": {
+    "question": "What's the population increase of Seattle from 2021 to 2023? Then check what's the stock price increase of Amazon from May 2023 to Feb 2023?",
+    "verbose": true
+  }
+}
+```
+Response shows agent runs `population_data_knowledge_base` and `stock_price_data_knowledge_base` tools:
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:722]:
```suggestion
The response shows that the agent runs both `population_data_knowledge_base` and `stock_price_data_knowledge_base` tools:
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 7]
```diff
@@ -0,0 +1,836 @@
...
+}
+```
+
+## 5. Configure root chatbot in OpenSearch Dashboard
+
+Read this [document](https://opensearch.org/docs/latest/ml-commons-plugin/opensearch-assistant/) for more details. 
+
+OpenSearch 2.12 released experimental Chabot UI on OpenSearch Dashboard. To use it, you need to configure a root Chatbot agent first.
+
+A root chatbot agent consists of
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:782]:
```suggestion
A root chatbot agent consists of the following parts:
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 8]
```diff
@@ -0,0 +1,836 @@
...
+ curl -k --cert ./kirk.pem --key ./kirk-key.pem -X PUT https://localhost:9200/.plugins-ml-config/_doc/os_chat -H 'Content-Type: application/json' -d'
+ {
+   "type":"os_chat_root_agent",
+   "configuration":{
+     "agent_id": "your_root_chatbot_agent_id"
+   }
+ }'
+```
+
+Go to your OpenSearch Dashboard config folder, `$OSD_HOME/config`, edit `opensearch_dashboards.yml` by adding this line to the end `assistant.chat.enabled: true`
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:828]:
```suggestion
Go to your OpenSearch Dashboards config folder, `$OSD_HOME/config`, edit `opensearch_dashboards.yml` by adding the following line to the end: `assistant.chat.enabled: true`.
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 9]
```diff
@@ -0,0 +1,836 @@
...
+
+Restart OpenSearch Dashboard, click the chat icon
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:830]:
```suggestion
Restart OpenSearch Dashboards, then select the chat icon, as shown in the following image.
```

File: docs/tutorials/agent_framework/build_your_own_chatbot.md [hunk 10]
```diff
@@ -0,0 +1,836 @@
...
+
+![Alt text](images/chatbot/osd_chatbot_1.png)
+
+Then chat on OpenSearch Dashboard
```
Comment (kolchfa-aws) [inline docs/tutorials/agent_framework/build_your_own_chatbot.md:834]:
```suggestion
You can now chat in OpenSearch Dashboards, as shown in the following image.
```

**Review by kolchfa-aws — COMMENTED** [review]
Some clarifying rewrites.

**Review by b4sjoo — APPROVED** [review]

**Pull request merged**